logger/simple.go

```diff
@@ -5,28 +5,45 @@ import (
 	"log"
 )
 
-type SimpleLogger struct{}
+type SimpleLogger struct {
+	Level Level
+}
 
 func (l *SimpleLogger) Enabled(_ context.Context, level Level) bool {
-	return true
+	return level >= l.Level
 }
 
-func (l *SimpleLogger) Tracef(_ context.Context, format string, v ...any) {
+func (l *SimpleLogger) Tracef(ctx context.Context, format string, v ...any) {
+	if !l.Enabled(ctx, l.Level) {
```

Comment:
```suggestion
	if !l.Enabled(ctx, l.LevelTrace) {
```

if not changed, it'll always return true ;) 

change it in other places as well

Reply:
The last two comments including this one are valid, but do you think we can add tests to validate them?